theano/scalar/basic.py

```diff
@@ -2658,6 +2658,13 @@ class Neg(UnaryScalarOp):
     nfunc_spec = ('negative', 1, 1)
 
     def impl(self, x):
+        # We have to make sure x is not a numpy.bool_, because
+        # `-numpy.bool_(True)` is `False` (we want 0), and
```

Comment:
numpy.bool_ is python bool. numpy.bool is a numpy scalar of type bool. 

So mostly, the problem is with python bool, not numpy bool.

It would be great to add a test for that. Do you still have the problematic case? If not, I'll try to make one.